### PR TITLE
Unify features generation and deployment

### PR DESCRIPTION
**Pull request by cynepco3hahue**
# Description

To ensure that make target will deploy your feature, you shoud:

- in case if you need to generate manifests before deploy, add generate.sh script under `features/<feature_name>/generate.sh`
- add deploy script under `features/<feature_name>/deploy.sh`
- add you feature under `Makefile`, see `export FEATURES`

You also can add and use some common variables that placed under `features/hack/common.sh`

## Type of change

Please select the appropriate options:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Refactoring

## Checklist

- [x] I have performed a self-review of my own code
- [x] If a change is adding a feature, it should require a change to the README.md and the review should catch this.
- [ ] If the change is a fix, it should have an issue. The review should make sure the comments state the issue (not just the number) and it should use the keywords that will close the issue on merge.
- [ ] A change should not be merged unless it passes CI or there is a comment/update saying what testing was passed.
- [ ] PRs should not be merged unless positively reviewed.

### CODE REVIEW

File: features/performance/manifests/templates/00-machine-config-pool-worker-rt.yaml.in [hunk 1]
```diff
@@ -0,0 +1,14 @@
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfigPool
+metadata:
+  name: worker-rt
+  labels:
+    worker-rt: ""
+spec:
+  machineConfigSelector:
+    matchExpressions:
+      - {key: machineconfiguration.openshift.io/role, operator: In, values: [worker,worker-rt]}
```
Comment (slintes) [inline features/performance/manifests/templates/00-machine-config-pool-worker-rt.yaml.in]: Both worker and worker-rt?
Doesn't this conflict with the default worker pool?

Comment (cynepco3hahue) [reply features/performance/manifests/templates/00-machine-config-pool-worker-rt.yaml.in]: as far as I saw no, and we can not update `KubeletConfig` without providing a default config file, see - https://bugzilla.redhat.com/show_bug.cgi?id=1772490

Comment (slintes) [reply features/performance/manifests/templates/00-machine-config-pool-worker-rt.yaml.in]: I see, thx :+1:

**Comment by cynepco3hahue** [comment]
@slintes @davidvossel @fedepaol Can you please take a look?

File: features/mcp/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+source $(dirname "$0")/../hack/common.sh
+
+oc apply -f ${MCP_DIR}/00-mcp-worker-rt.yaml
```
Comment (fedepaol) [inline features/mcp/deploy.sh]: if we want to tie this to func tests, we must ensure that deploy also waits for the deployment to be ready.
Once we have that, we'll be able to run ` export FEATURES="feat1 feat2" && make deploy && make functests` (see https://github.com/openshift-kni/baremetal-deploy/pull/66 )

Comment (fedepaol) [reply features/mcp/deploy.sh]: Never mind, I thought it was a machine config. Ignore my previous comment

File: features/Makefile [hunk 3]
```diff
@@ -1,25 +1,37 @@
...
-deploy:
-	for f in $(FEATURES) ; \
-    do \
-	  echo "deploying $$f"; \
-      ./$$f/deploy.sh; \
-    done \
+.PHONY: deploy
+deploy: generate
+	for f in $(FEATURES); do \
+		if [ -f ./$$f/deploy.sh ]; then \
```
Comment (fedepaol) [inline features/Makefile]: Neat!

**Comment by slintes** [comment]
nice :+1: 
/lgtm

File: features/Makefile (hunk 3 above)
Comment (MarSik) [reply features/Makefile]: Case sensitive though.. and I would add else with echo "WARN: No such feature $f" or something

Comment (cynepco3hahue) [reply features/Makefile]: will add the WARN message

**Comment by fedepaol** [comment]
> /lgtm
> /override ci/prow/sctp-e2e

sctp-e2e should be back to normal btw

File: features/Makefile (hunk 3 above)
Comment (cynepco3hahue) [reply features/Makefile]: done

**Comment by cynepco3hahue** [comment]
> sctp-e2e should be back to normal btw

Great, let see how it will work.

**Comment by davidvossel** [comment]
@cynepco3hahue   very nice PR :+1:

**Pull request merged**